### PR TITLE
Add accessibility help menu to chat input 

### PR DESCRIPTION
**Pull request by meganrogge**
Part of https://github.com/microsoft/vscode-internalbacklog/issues/4109

Aligns with the terminal accessibility help menu  experience and adds a setting so this can be disabled for power users

Reuses the input box, just replacing the text to reduce complexity & code while enabling the screen reader user to stay within their current context.

I discussed doing the same thing - with different help text content - for the interactive session input box in the editor with @jrieken

### CODE REVIEW

File: src/vs/workbench/contrib/accessibility/browser/accessibilityContribution.ts [hunk 1]
```diff
@@ -23,6 +23,12 @@ const configuration: IConfigurationNode = {
 			type: 'boolean',
 			default: true,
 			tags: ['accessibility']
+		},
+		'accessibility.verbosity.interactiveSessionInput': {
```
Comment (roblourens) [inline src/vs/workbench/contrib/accessibility/browser/accessibilityContribution.ts:27]: call it `.chatInput`, I'm going to transition to that naming

File: src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionAccessibilityHelp.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import { localize } from 'vs/nls';
+import { format } from 'vs/base/common/strings';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+
+export function getAccessibilityHelpText(keybindingService: IKeybindingService): string {
+	const content = [];
+	content.push(localize('interactiveSession.accessibilityHelp', "Interactive Session Accessibility Help"));
+	content.push(localize('interactiveSession.helpMenuExit', "Exit this menu and return to the interactive editor input via the Escape key."));
+	content.push(descriptionForCommand('interactiveSession.action.focus', localize('interactiveSession.action.focus', 'The Focus Interactive Session ({0}) command focuses the chat request/response list, which can be navigated with UpArrow/DownArrow.'), localize('interactiveSession.action.focusNoKb', 'The Focus Interactive Session command focuses the chat request/response list, which can be navigated with UpArrow/DownArrow and is currently not triggerable by a keybinding.'), keybindingService));
```
Comment (roblourens) [inline src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionAccessibilityHelp.ts:14]: Missing `{0}` parameter?

File: src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionActions.ts [hunk 3]
```diff
@@ -124,6 +128,57 @@ export function registerInteractiveSessionActions() {
 		}
 	});
 
+	registerEditorAction(class FocusInteractiveSessionAction extends EditorAction {
+		constructor() {
+			super({
+				id: 'interactiveSession.action.accessibilityHelp',
+				label: localize('actions.interactiveSession.accessibiltyHelp', "Interactive Session Accessibility Help"),
+				alias: 'Focus Interactive Session',
```
Comment (roblourens) [inline src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionActions.ts:136]: Shouldn't the alias and class name be about "accessibility help"?

File: src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionActions.ts [hunk 4]
```diff
@@ -124,6 +128,57 @@ export function registerInteractiveSessionActions() {
 		}
 	});
 
+	registerEditorAction(class FocusInteractiveSessionAction extends EditorAction {
+		constructor() {
+			super({
+				id: 'interactiveSession.action.accessibilityHelp',
+				label: localize('actions.interactiveSession.accessibiltyHelp', "Interactive Session Accessibility Help"),
```
Comment (roblourens) [inline src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionActions.ts:135]: Use the category that other actions use

File: src/vs/workbench/contrib/interactiveSession/browser/interactiveSession.ts [hunk 5]
```diff
@@ -37,8 +37,7 @@ export interface IInteractiveSessionWidget {
 	readonly viewModel: IInteractiveSessionViewModel | undefined;
 	readonly inputEditor: ICodeEditor;
 	readonly providerId: string;
-
```
Comment (roblourens) [inline src/vs/workbench/contrib/interactiveSession/browser/interactiveSession.ts:40]: Is this change on purpose?

File: src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts [hunk 6]
```diff
@@ -123,15 +141,19 @@ export class InteractiveSessionInputPart extends Disposable implements IHistoryN
 		this._inputEditor.focus();
 	}
 
-	async acceptInput(query?: string | IInteractiveSessionReplyFollowup): Promise<void> {
+	async acceptInput(query?: string | IInteractiveSessionReplyFollowup, skipExecution?: boolean): Promise<void> {
```
Comment (roblourens) [inline src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts:144]: The input editor is already exposed, and you can operate on that directly. I don't think we should add this parameter that totally changes the meaning of the method

File: src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionViewPane.ts [hunk 7]
```diff
@@ -132,7 +133,9 @@ export class InteractiveSessionViewPane extends ViewPane implements IInteractive
 		this._widget.saveState();
 
 		const widgetViewState = this._widget.getViewState();
-		this.viewState.inputValue = widgetViewState.inputValue;
+
+		// do not save input if it is the accessibility help text
```
Comment (roblourens) [inline src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionViewPane.ts:137]: Can you change the widget's `getViewState` to have this logic instead?

**Review by roblourens — CHANGES_REQUESTED** [review]

**Comment by roblourens** [comment]
Thanks!

File: src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionActions.ts (hunk 4 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionActions.ts:135]: this is an EditorAction, which does not have a `category`.

File: src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionAccessibilityHelp.ts (hunk 2 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionAccessibilityHelp.ts:14]: no - a bit confusing, but that get's set in `descriptionForCommand`
Screenshot 2023-05-12 at 2 40 11 PM

File: src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts [hunk 8]
```diff
@@ -129,9 +147,7 @@ export class InteractiveSessionInputPart extends Disposable implements IHistoryN
 			// Followups and programmatic messages don't go to history
 			this.history.add(editorValue);
 		}
-
 		this._inputEditor.focus();
-		this._inputEditor.setValue('');
```
Comment (roblourens) [inline src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts:134]: Is this supposed to be deleted?

Comment (meganrogge) [reply src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts:134]: I deleted it accidentally, but now I'm curious abt what that is meant to do

Comment (meganrogge) [reply src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts:134]: still works without it 🤔

Comment (meganrogge) [reply src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts:134]: pushed a commit to revert that, assuming it does something important

Comment (roblourens) [reply src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts:134]: It's supposed to clear the input editor when you submit a request. There is still a random deleted line here, it's a small thing but I don't want to remove random unrelated lines

File: src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionActions.ts [hunk 9]
```diff
@@ -124,6 +128,57 @@ export function registerInteractiveSessionActions() {
...
+					inputEditor.setValue(cachedInput);
+					inputEditor.setPosition(cachedPosition!);
+					widget.focusInput();
+					disposable.dispose();
+				}
+			});
+		}
+	});
+
+
```
Comment (meganrogge) [inline src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionActions.ts:181]:
```suggestion
```

File: src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts [hunk 10]
```diff
@@ -129,7 +147,6 @@ export class InteractiveSessionInputPart extends Disposable implements IHistoryN
 			// Followups and programmatic messages don't go to history
 			this.history.add(editorValue);
 		}
-
 		this._inputEditor.focus();
```
Comment (meganrogge) [inline src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts:150]:
```suggestion

		this._inputEditor.focus();
```

File: src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts (hunk 8 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/interactiveSession/browser/interactiveSessionInputPart.ts]: sorry for the back and forth. think it's fixed now

**Review by roblourens — APPROVED** [review]
